src/braket/tasks/gate_model_quantum_task_result.py

```diff
@@ -278,13 +279,25 @@ def _from_object_internal_computational_basis_sampling(cls, result: GateModelTas
                 f"Measured qubits {measured_qubits} is not equivalent to number of qubits "
                 + f"{measurements.shape[1]} in measurements"
             )
-        result_types = (
-            result.resultTypes
-            if result.resultTypes
-            else GateModelQuantumTaskResult._calculate_result_types(
+        if result.resultTypes:
+            # Jaqcd does not return anything in the resultTypes schema field since the
+            # result types are easily parsable from the IR. However, an OpenQASM program
+            # specifies result types inline and parsing result types is more involved
+            # (ie. may involve dereferencing logical qubits at runtime), so the parsed
+            # result type specifications need to be returned, even if not calculated
+            # during simulation.
+            if not isinstance(result.resultTypes[0], ResultTypeValue):
```

Comment:
Are we calculating the result types in the SDK even for OpenQASM ?

Reply:
Yes (for shots > 0), that is how it is currently implemented, in line with other BDK/Sim code, but slightly different from the service. If we ever want to shift to calculating all result types in the sim, it should be a non-breaking change

Reply:
Why would we not want to align the local sim & service behavior for OpenQASM now ?

Reply:
Right now, that part of the code follows the same computation paths for OpenQASM and Jaqcd and introducing different behavior would increase the complexity. Practically, I think it would make us yellow for the launch date at best, but if you think there is a compelling reason to do so, I can see if I can get it working and tested over the weekend. I'd be inclined to simplify and leave it as is for now

Reply:
Especially since anyone using the `LocalSimulator` class in the BDK will not see a difference